definitions/npm/styled-components_v4.x.x/flow_v0.92.x-/styled-components_v4.x.x.js

```diff
@@ -0,0 +1,410 @@
+    tfoot:                    StyledComponentType<StyledElementType<'tfoot'>, Theme>,
+    th:                       StyledComponentType<StyledElementType<'th'>, Theme>,
+    thead:                    StyledComponentType<StyledElementType<'thead'>, Theme>,
+    time:                     StyledComponentType<StyledElementType<'time'>, Theme>,
+    title:                    StyledComponentType<StyledElementType<'title'>, Theme>,
+    tr:                       StyledComponentType<StyledElementType<'tr'>, Theme>,
+    track:                    StyledComponentType<StyledElementType<'track'>, Theme>,
+    u:                        StyledComponentType<StyledElementType<'u'>, Theme>,
+    ul:                       StyledComponentType<StyledElementType<'ul'>, Theme>,
+    var:                      StyledComponentType<StyledElementType<'var'>, Theme>,
+    video:                    StyledComponentType<StyledElementType<'video'>, Theme>,
+    wbr:                      StyledComponentType<StyledElementType<'wbr'>, Theme>,
+
+    // SVG
+    circle:                   StyledComponentType<StyledElementType<'circle'>, Theme>,
+    clipPath:                 StyledComponentType<StyledElementType<'clipPath'>, Theme>,
+    defs:                     StyledComponentType<StyledElementType<'defs'>, Theme>,
+    ellipse:                  StyledComponentType<StyledElementType<'ellipse'>, Theme>,
+    g:                        StyledComponentType<StyledElementType<'g'>, Theme>,
+    image:                    StyledComponentType<StyledElementType<'image'>, Theme>,
+    line:                     StyledComponentType<StyledElementType<'line'>, Theme>,
+    linearGradient:           StyledComponentType<StyledElementType<'linearGradient'>, Theme>,
+    mask:                     StyledComponentType<StyledElementType<'mask'>, Theme>,
+    path:                     StyledComponentType<StyledElementType<'path'>, Theme>,
+    pattern:                  StyledComponentType<StyledElementType<'pattern'>, Theme>,
+    polygon:                  StyledComponentType<StyledElementType<'polygon'>, Theme>,
+    polyline:                 StyledComponentType<StyledElementType<'polyline'>, Theme>,
+    radialGradient:           StyledComponentType<StyledElementType<'radialGradient'>, Theme>,
+    rect:                     StyledComponentType<StyledElementType<'rect'>, Theme>,
+    stop:                     StyledComponentType<StyledElementType<'stop'>, Theme>,
+    svg:                      StyledComponentType<StyledElementType<'svg'>, Theme>,
+    text:                     StyledComponentType<StyledElementType<'text'>, Theme>,
+    tspan:                    StyledComponentType<StyledElementType<'tspan'>, Theme>
+  }
+
+  declare export default StyledComponentList & {
+    [[call]]: <S : string>(S) => $ElementType<StyledComponentList, S>,
+    [[call]]: <T: Theme, Props: {}, C: React$ComponentType<Props>>(C) => StyledComponentType<React$ComponentType<{...$Exact<Props>, className?: string}>, T>
```

Comment:
There are a few warnings that remain in the tests, that are due to this line, as best as I can tell. Editing this line as follows removes the warnings:
```suggestion
    [[call]]: <T: Theme, Props: {}, C: React$ComponentType<Props>>(C) => StyledComponentType<React$ComponentType<Props>, T>
```

But even editing this line to below still causes the warnings:
```suggestion
    [[call]]: <T: Theme, Props: {}, C: React$ComponentType<Props>>(C) => StyledComponentType<React$ComponentType<$Exact<Props>>, T>
```

I don't understand why this last example still causes errors. I don't see how `$Exact<Props>` should behave any differently than `Props` for the component type (or at least, I would expect it to cause _additional_ errors to arise, not fewer). Can anyone shed some light into why this isn't working as expected?